sources/engine/Stride.Physics/Engine/PhysicsConstraintProcessor.cs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using Stride.Core;
+using Stride.Core.Annotations;
+using Stride.Core.Collections;
+using Stride.Core.Diagnostics;
+using Stride.Engine;
+using Stride.Games;
+
+namespace Stride.Physics.Engine
+{
+    public class PhysicsConstraintProcessor : EntityProcessor<PhysicsConstraintComponent>
+    {
+        private static readonly Logger logger = GlobalLogger.GetLogger(nameof(PhysicsConstraintProcessor));
+
+        private readonly FastList<PhysicsConstraintComponent> detachedComponents = new FastList<PhysicsConstraintComponent>();
+
+        public PhysicsConstraintProcessor()
+        {
+            Order = 0xFFFF; // After PhysicsProcessor
+        }
+
+        protected override void OnEntityComponentAdding(Entity entity, [NotNull] PhysicsConstraintComponent component, [NotNull] PhysicsConstraintComponent data)
+        {
+            component.ConstraintProcessor = this;
+            component.Detached = false;
+
+            // this is mostly required for the game studio gizmos
+            if (Simulation.DisableSimulation)
+            {
+                return;
+            }
+
+            Recreate(component, skipUninitializedComponents: true);
+        }
+
+        protected override void OnEntityComponentRemoved(Entity entity, [NotNull] PhysicsConstraintComponent component, [NotNull] PhysicsConstraintComponent data)
+        {
+            DisposeOf(component);
+        }
+
+        public override void Update(GameTime time)
+        {
+            // this is mostly required for the game studio gizmos
+            if (Simulation.DisableSimulation)
+                return;
+
+            detachedComponents.Clear();
+
+            foreach (var datas in ComponentDatas)
+            {
+                var component = datas.Key;
+
+                if (component.Constraint != null)
+                {
+                    if (component.Constraint.InternalConstraint != null)
+                    {
+                        component.Constraint.Enabled = component.Enabled;
+                    }
+                    else
+                    {
+                        // the constraint is set but also disposed, let's clear it
+                        DisposeOf(component);
+                        detachedComponents.Add(component);
+                    }
+                }
+                else if (component.Constraint == null && component.Enabled)
+                {
+                    Recreate(component, component.Detached);
+                }
+            }
+
+            foreach (var component in detachedComponents)
+            {
+                component.OnDetach();
+            }
```

Comment:
``detachedComponents.Clear`` right here instead to avoid holding refs within that list for longer than necessary (or 'forever' for cases where things have been detached in the previous frame but the current has ``Simulation.DisableSimulation`` true which would prevent clear and the rest of update to execute)